trieste/models/gpflux/sampler.py

```diff
@@ -398,44 +441,105 @@ def __init__(self, layer: GPLayer, n_components: int):
 
         if isinstance(layer.kernel, gpflow.kernels.SharedIndependent):
             self._kernel = layer.kernel.kernel
+        elif isinstance(layer.kernel, gpflow.kernels.SeparateIndependent):
+            self._kernel = layer.kernel.kernels
         else:
             self._kernel = layer.kernel
         self._n_components = n_components
-        super().__init__(self._kernel, self._n_components, dtype=tf.float64)
 
-        if isinstance(layer.inducing_variable, InducingPoints):
-            inducing_points = layer.inducing_variable.Z
+        if isinstance(layer.inducing_variable, gpflow.inducing_variables.SharedIndependentInducingVariables):
+            self._inducing_points = layer.inducing_variable.Z # [M, D]
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                Kuf(layer.inducing_variable, layer.kernel, x)
+            ) # [M, N]
+            dummy_X = self._inducing_points[0:1, :]
+
+        elif isinstance(layer.inducing_variable, gpflow.inducing_variables.SeparateIndependentInducingVariables):
+            self._inducing_points = tf.stack([ind_var.Z for ind_var in layer.inducing_variable.inducing_variable_list], axis = 0) # [P, M, D]
+
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                Kuf(layer.inducing_variable, layer.kernel, x)
+            ) # [P, M, N]
+
+            dummy_X = self._inducing_points[0][0:1, :]
+
         else:
-            inducing_points = layer.inducing_variable.inducing_variable.Z
+            self._inducing_points = layer.inducing_variable.inducing_variable.Z  # [M, D]
+            self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
+                Kuf(layer.inducing_variable, layer.kernel, x)
+            ) # [M, N]
+            dummy_X = self._inducing_points[0:1, :] 
 
-        self._canonical_feature_functions = lambda x: tf.linalg.matrix_transpose(
-            self._kernel.K(inducing_points, x)
-        )
 
-        dummy_X = inducing_points[0:1, :]
+        # Build the RFF objects
+        if isinstance(self._kernel, list):
+
+            self._rff = defaultdict()
```

Comment:
you'll need to provide a type here most likely (see below), though not sure why you need `defaultdict` here, it seems like you can simply do `dict()`...
```
self._rff: DefaultDict[int, RFF] = defaultdict()
```

Reply:
added it